core/domain/opportunity_domain.py

```diff
@@ -198,3 +198,97 @@ def validate(self):
                 'incomplete_translation, needs_voiceover and assigned_voiceover'
                 ' to be the same as the supported audio languages, '
                 'received %s' % list(expected_set_of_all_languages))
+
+
+class SkillOpportunity(python_utils.OBJECT):
+    """The domain object for skill opportunities."""
+
+    def __init__(
+            self, skill_id, skill_description,
+            question_count=0, topic_id=None, topic_name=None):
+        """Constructs a SkillOpportunity domain object.
+
+        Args:
+            skill_id: str. The unique id of the skill.
+            skill_description: str. The title of the skill.
+            question_count: int. The total number of questions for the skill.
+            topic_id: str. The unique id of the topic.
+            topic_name: str. The name of the topic.
+        """
+        self.topic_id = topic_id
+        self.topic_name = topic_name
+        self.skill_id = skill_id
+        self.skill_description = skill_description
+        self.question_count = question_count
+        self.validate()
+
+    def validate(self):
+        """Validates various properties of the object.
+
+        Raises:
+            ValidationError: One or more attributes of the object are invalid.
+        """
+        if (not isinstance(self.topic_id, python_utils.BASESTRING) and
+                self.topic_id is not None):
```

Comment:
Not sure that topic_id = None is actually a valid case. If we're surfacing a skill in the opportunity dashboard, shouldn't it already be linked to a topic? /cc @DubeySandeep

Reply:
Looking at the topic and skills dashboard, it appeared to me that skills could exist unlinked to a specific topic, so I was operating under that assumption here. I was hoping to get clarification by the code owners as well.

Reply:
Skills can exist unlinked to a specific topic, but we don't care about showing those in the opportunity dashboard. With regards to the lifecycle of the opportunity objects, I think that's a design question to discuss with @DubeySandeep.